go/ct/st/state.go

```diff
@@ -400,5 +421,26 @@ func (s *State) Diff(o *State) []string {
 		res = append(res, fmt.Sprintf("Different return data: %x vs %x", s.ReturnData, o.ReturnData))
 	}
 
+	if s.HasSelfDestructed != o.HasSelfDestructed {
+		res = append(res, fmt.Sprintf("Different has-self-destructed: %v vs %v ", s.HasSelfDestructed, o.HasSelfDestructed))
+	}
+
+	if !slices.Equal(s.SelfDestructedJournal, o.SelfDestructedJournal) {
+		if len(s.SelfDestructedJournal) != len(o.SelfDestructedJournal) {
+			res = append(res, fmt.Sprintf("Different has-self-destructed journal length: %v vs %v",
+				len(s.SelfDestructedJournal), len(o.SelfDestructedJournal)))
+		} else {
+			for index, entry1 := range s.SelfDestructedJournal {
+				account1, beneficiary1 := entry1.account, entry1.beneficiary
+				entry2 := o.SelfDestructedJournal[index]
+				account2, beneficiary2 := entry2.account, entry2.beneficiary
+				if !reflect.DeepEqual(account1, account2) || !reflect.DeepEqual(beneficiary1, beneficiary2) {
+					res = append(res, fmt.Sprintf("Different has-self-destructed journal entry:\n\t(%v, %v)\n\tvs\n\t(%v, %v)",
+						account1, beneficiary1, account2, beneficiary2))
+				}
```

Comment:
The accounts can be compared with ==, and as a consequence also the entries. So this can be reduced to
```suggestion
				entry2 := o.SelfDestructedJournal[index]
				if entry1 != entry2 {
					res = append(res, fmt.Sprintf("Different has-self-destructed journal entry:\n\t(%v, %v)\n\tvs\n\t(%v, %v)",
						entry1.account, entry1.beneficiary, entry2.account, entry2.beneficiary))
				}
```